test/pytest/receiver/test_receiver.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from pathlib import Path
+import sys
+import time
+import pytest
+
+receiver_path = (
+    Path(__file__).parent.parent.parent.parent / "src/hidra/receiver")
+assert receiver_path.is_dir()
+sys.path.insert(0, receiver_path)
+
+from datareceiver import PluginHandler, run_plugin_thread  # noqa
+import plugins.mock_plugin as mock_plugin_m  # noqa
+
+log = logging.getLogger(__name__)
+
+metadata = {
+    "relative_path": "relative_path",
+    "filename": "filename"}
+
+
+@pytest.fixture
+def mock_plugin():
+    mock_plugin_m.Plugin.side_effect = None
+    mock_plugin = mock_plugin_m.Plugin(None)
+    mock_plugin_m.Plugin.reset_mock()
+    mock_plugin.reset_mock()
+    mock_plugin.setup.side_effect = None
+    mock_plugin.process.side_effect = None
+    mock_plugin.stop.side_effect = None
+    return mock_plugin
+
+
+def test_plugin_start_stop(mock_plugin):
+    plugin_handler = PluginHandler("mock_plugin", {"foo": 1}, "bar", log)
+    plugin_handler.start()
+    plugin_handler.stop()
+    mock_plugin_m.Plugin.assert_called_once_with({"foo": 1})
+    mock_plugin = mock_plugin_m.Plugin(None)
+    mock_plugin.setup.assert_called_once_with()
+    mock_plugin.stop.assert_called_once_with()
+    mock_plugin.process.assert_not_called()
+
+
+def test_plugin_put(mock_plugin):
+    plugin_handler = PluginHandler("mock_plugin", {"foo": 1}, "bar", log)
+    plugin_handler.start()
+    plugin_handler.put((metadata, 1))
+    plugin_handler.stop()
+    mock_plugin.process.assert_called_once_with(
+        local_path="bar/relative_path/filename",
```

Comment:
isn't it possible to fix this as a variable, instead of writing it manually?

Reply:
I could write
```
def test_plugin_put(mock_plugin):
    target_dir = "bar"
    plugin_handler = PluginHandler("mock_plugin", {"foo": 1}, target_dir, log)
    plugin_handler.start()
    plugin_handler.put((metadata, 1))
    plugin_handler.stop()
    local_path = os.path.join(target_dir, metadata["relative_path"], metadata["filename"])
    mock_plugin.process.assert_called_once_with(
        local_path=local_path,
        metadata=metadata,
        data=1)
```
but I think this would make the test less readable without a clear advantage.

Reply:
sorry, that was misleading. I only meant the single line of assignment test of 'local_path'